gameros/airootfs/root/install.sh

```diff
@@ -2,46 +2,55 @@
 
 
 if [ $EUID -ne 0 ]; then
-	echo "$(basename $0) must be run as root"
-	exit 1
+    echo "$(basename $0) must be run as root"
+    exit 1
 fi
 
-#### Internet connection detection ####
-dhcpcd --background
-
+#### Wait for conenction or ask the user for configuration ####
 whiptail --infobox "Checking connection..." 10 50
 sleep 5
 
 while ! ( curl -Is https://gamer-os.github.io/ | head -1 | grep 200 > /dev/null ); do
-	whiptail --yesno "No internet connection detected. Please connect this computer\
-	 to the internet with a wired connection, wait a few seconds, then retry." 10 50\
-	 --yes-button "Retry"\
-	 --no-button "Exit"
-
-	if [ $? -ne 0 ]; then
-		 exit 1
-	fi
+    whiptail --yesno "No wired connection detected. Please connect this computer \
+     to the internet by configuring a new network." 10 50 \
+     --yes-button "Configure" \
+     --no-button "Exit"
+
+    if [ $? -ne 0 ]; then
+         exit 1
+    fi
+
+    nmtui
 done
 #######################################
 
 #### Install frzr
 curl -L https://github.com/gamer-os/frzr/releases/download/0.7.0/frzr-0.7.0-1-any.pkg.tar.zst > frzr.pkg.tar.zst
 pacman --noconfirm -U frzr.pkg.tar.zst
 if [ $? -ne 0 ]; then
-	exit 1
+    exit 1
 fi
 rm frzr.pkg.tar.zst
 
 if ! frzr-bootstrap gamer; then
-	exit
+    exit 1
+fi
+
+# Post install steps for system configuration
+# Copy over all network configuration from the live session to the system
+MOUNT_PATH=/tmp/frzr_root
+if [ -d "/etc/NetworkManager/system-connections" ]; then
```

Comment:
I really don't like this step being here. I think it's much better to have this in frzr.

Reply:
I talked to @alkazar and he thinks it's better here for now.
My best proposed solution is having something like
````
if [ -f "frzr-postinstall" ]; then
 frzr-postinstall
fi
````
In `frzr-bootstrap` as a hook for any installer to use. Then the post install steps are the frzr user (the install media) responsibility to use just by adding that script. That way is generic on the frzr side and agnostic of the installer.
The installer is more tied to the image and machine configuration so it should be the one who prepares those install steps.

Anyway is a matter of preference and right now has the same effect.

Reply:
Ah okay, I see. I took a look at the frzr codebase and it has changed quite a bit from what I remember. Looks like this is the right place after all.

Reply:
You can close this comment